include/openthread/dnssd_server.h

```diff
@@ -178,6 +197,29 @@ void otDnssdQueryHandleDiscoveredServiceInstance(otInstance *                aIn
  */
 void otDnssdQueryHandleDiscoveredHost(otInstance *aInstance, const char *aHostFullName, otDnssdHostInfo *aHostInfo);
 
+/**
+ * This function aquires the next query in the DNS-SD server.
+ *
+ * @param[in] aInstance         The OpenThread instance structure.
+ * @param[in] aQuery            The query pointer. Pass NULL to get the first query.
+ *
+ * @returns  A pointer to the query or NULL if no more queries.
+ *
+ */
+const otDnssdQuery *otDnssdGetNextQuery(otInstance *aInstance, const otDnssdQuery *aQuery);
+
+/**
+ * This function aquires the DNS-SD query type and name for a specific query.
+ *
+ * @param[in]   aQuery            The query pointer acquired from `otDnssdGetNextQuery`.
+ * @param[out]  aNameOutput       The name output buffer, recommended be at least 255 bytes long.
```

Comment:
```suggestion
 * @param[out]  aNameOutput       The name output buffer, which should be `OT_DNS_MAX_NAME_SIZE` bytes long.
```
By `char (*aNameOutput)[OT_DNS_MAX_NAME_SIZE]`, the buffer has to be `OT_DNS_MAX_NAME_SIZE` long, right?

Reply:
I don't think so, array in C laungage is just pointer, there is no length properties of a array type.